### ISSUE TITLE
Colorized bullet point

### ISSUE THREAD

**Issue by anotherduckling** [retypeapp/retype#603]
Is it possible to add colorized bullet point (for both unsorted types)? It will be easier and will take less space than other method like using header or column. Will be helpful. 

![image](https://github.com/retypeapp/retype/assets/57977673/d28b4888-df24-4719-be1a-7dc602bb2e1d)

**Comment by geoffreymcgill** [comment]
It is possible, although not super clean. The following sample demonstrates one technique:

```html
<style>
#list-1 li:nth-child(1)::marker,
#list-1 li:nth-child(2)::marker {
    color: green;
}

#list-1 li:nth-child(3)::marker {
    color: yellow;
}

#list-1 li:nth-child(4)::marker {
    color: red;
}
</style>

{#list-1}
- Pros 1
- pros 2
- keep in mind
- con
```

The above creates the following results:

<img width="291" alt="Screenshot 2023-08-12 at 4 22 29 AM" src="https://github.com/retypeapp/retype/assets/62210/c37bd315-de13-49db-b525-e50531b752de">

The big disadvantage of the above technique is that the CSS would need to be kept in sync with the bullet list. If the bullet list changes, the `<style>` would have to be updated. 

Adding items to the bottom of the list would be a little easier than inserting a new list item somewhere higher. If adding a list item to the bottom, then just add another CSS spec. For instance, the following sample demonstrates adding a 5th `blue` item by using the `nth-child(5)` specifier:

```css
#list-1 li:nth-child(5)::marker {
    color: blue;
}
```

<img width="314" alt="Screenshot 2023-08-12 at 4 36 42 AM" src="https://github.com/retypeapp/retype/assets/62210/8c800c57-6367-4e51-b5e2-f012d11d9fc6">

You could also change the default color of the markers (bullets). The following sample demonstrates changing the default color of all bullets to `green`, then you only need to explicitly define the positions of the non-default colored list items. 

```html
<style>
#list-1 li::marker {
    color: green;
}

#list-1 li:nth-child(3)::marker {
    color: yellow;
}

#list-1 li:nth-child(4)::marker {
    color: red;
}
</style>
```

This creates the same result as the first sample:

<img width="291" alt="Screenshot 2023-08-12 at 4 22 29 AM" src="https://github.com/retypeapp/retype/assets/62210/c37bd315-de13-49db-b525-e50531b752de">

Instead of maintaining the `<style>` block inside the `.md` page file, it is possible to add to the `_includes/head.html` using the techniques outlined in the following response:

https://github.com/retypeapp/retype/issues/600#issuecomment-1673770090

---

Theoretically, it would be possible for Retype to support the following syntax, which would avoid having to maintain the `nth-child(#)` positions. Although unfortunately, the following syntax is not currently supported. 

```html
<style>
li.item-green::marker {
    color: green;
}

li.item-yellow::marker {
    color: yellow;
}

li.item-red::marker {
    color: red;
}
</style>

- Pros 1 {.item-green}
- pros 2 {.item-green}
- keep in mind {.item-yellow}
- con {.list-red}
```

I have created a feature request to support the `- Pros 1 {.item-green}` generic attribute syntax within Retype, although I do not have a clear timeline on when we could build this functionality.

Hope this helps.

**Comment by geoffreymcgill** [comment]
Another option to consider is adding an [icon](https://retype.com/components/icon/) to each list item.

```md
- :icon-check-circle: Pros 1
- :icon-check-circle: pros 2
- :icon-alert: keep in mind
- :icon-circle-slash: con
```

The above sample creates the following list:

<img width="352" alt="Screenshot 2023-08-12 at 4 43 18 AM" src="https://github.com/retypeapp/retype/assets/62210/b62e27fd-d8d9-4f2f-882f-597c4913f896">

You could take this even further by using a [`[!badge]`](https://retype.com/components/badge/) component.

```md
- [!badge variant="info" text=":icon-check-circle:"] Pros 1
- [!badge variant="info" text=" :icon-check-circle:"] pros 2
- [!badge variant="info" text=" :icon-alert:"] keep in mind
- [!badge variant="info" text=" :icon-circle-slash:"] con
```

The above sample creates the following list:

<img width="367" alt="Screenshot 2023-08-12 at 4 43 29 AM" src="https://github.com/retypeapp/retype/assets/62210/268caaf6-590c-4ad1-b49c-095954eded2c">

You could also experiment with changing the [`variant`](https://retype.com/components/badge/#variant) on each of the badges.

**Comment by geoffreymcgill** [comment]
I thought of another technique. Using CSS to remove the bullet point.

```
<style>
#list-2 {
    list-style-type: none;
    padding-left: 1rem;
}
</style>

{#list-2}
- :large_green_circle: Pros 1
- :large_green_circle: pros 2
- :large_yellow_circle: keep in mind
- :red_circle: con
```

<img width="372" alt="Screenshot 2023-08-12 at 5 13 42 AM" src="https://github.com/retypeapp/retype/assets/62210/d07b304c-d915-4ec3-8734-fc22b8e4af7e">

You could add a little more space to the right of the emoji by using the `nbsp;` work around.

```
{#list-2}
- :large_green_circle: &nbsp;Pros 1
- :large_green_circle: &nbsp;pros 2
- :large_yellow_circle: &nbsp;keep in mind
- :red_circle: &nbsp;con
```

<img width="389" alt="Screenshot 2023-08-12 at 5 12 55 AM" src="https://github.com/retypeapp/retype/assets/62210/ffa2ecac-b75f-4a1f-9689-665d596bf979">

This technique also works well with icons.

```
<style>
#list-2 {
    list-style-type: none;
    padding-left: 1rem;
}
</style>

{#list-2}
- :icon-check-circle: &nbsp;Pros 1
- :icon-check-circle: &nbsp;pros 2
- :icon-alert: &nbsp;keep in mind
- :icon-circle-slash: &nbsp;con
```

<img width="382" alt="Screenshot 2023-08-12 at 5 17 52 AM" src="https://github.com/retypeapp/retype/assets/62210/02dff971-2cee-4d39-80b9-594ea251c77b">

Hope this helps.

**Comment by anotherduckling** [comment]
thanks. what will be the code for sub category? 
```
<style>
#icon-list {
    list-style-type: none;
    padding-left: 1rem;
}
</style>

{#icon-list}
- [**KickAssAnime**](URL) [!button variant="primary" size="xs" text="Proxies"](URL)
    - :icon-check-circle: &nbsp;Scrapes **Crunchyroll**(duckstream) and **Bilibili**(birdstream)
    - :icon-alert: &nbsp;KAA encode(videstreaming) is ok-ish with lower overall bitrate. Use the previous two if available. Duck has the highest overall bitrate
    - :icon-check-circle: &nbsp;Has both hardsub(duck) & softsub(rest)
    - :icon-circle-slash: &nbsp;Still a beta site and has missing titles. But, you can request them in their support server
 ```
    
    
![image](https://github.com/retypeapp/retype/assets/57977673/bf3dd1fa-56db-43b3-8f3a-2372b73e9eb9)

**Comment by geoffreymcgill** [comment]
> thanks. what will be the code for sub category?

The `<style>` would be:

```html
<style>
#icon-list, #icon-list ul {
    list-style-type: none;
    padding-left: 1rem;
}
</style>
```

Hope this helps.

**Comment by anotherduckling** [comment]
Thanks! But what will be the style code for subcategory? I have tried this but didn't remove the bullets for subcategory

```
<style>
#icon-list {
    list-style-type: none;
    padding-left: 1rem;
}
</style>

{#icon-list}
- Product1
    - :icon-check-circle: &nbsp;Pro
    - :icon-alert: &nbsp;Keep in mind
    - :icon-circle-slash: &nbsp;Con
```

![image](https://github.com/retypeapp/retype/assets/57977673/4d320451-4294-4fee-8db9-cdb10b63539f)

**Comment by anotherduckling** [comment]
thanks a lot. it has solved the issue.

**Comment by geoffreymcgill** [comment]
As of Retype v3.3, we have added built in support for a new `list-icon` CSS class, so you will not have to add your own `<style>` block as mentioned above. 

As well, the new `list-icon` class also supports automatically adding the correct padding to the right of the icon, so you will not need to use the `&nbsp;` work-around I proposed above.

```md
# Before
- :icon-check-circle: &nbsp;Pros 1

# After
- :icon-check-circle: Pros 1
```

To create an icon list, just apply the `.list-icon` css class using the following technique:

```md
{.list-icon}
- :icon-alert: Item 1
    - :icon-alert: This is sub-item 1.1
    - :icon-circle-slash: This is sub-item 1.2
- Item 2
    - :icon-check-circle: This is sub-item 2.1
    - :icon-check-circle: This is sub-item 2.2
    - :icon-alert: This is sub-item 2.3
    - :icon-circle-slash: This is sub-item 2.4
```

The list above sample will be converted into the following icon list:

<img width="444" alt="Screen Shot 2023-08-24 at 9 57 28 PM" src="https://github.com/retypeapp/retype/assets/62210/8c625854-ad46-48c4-9f79-bca3d7a411da">

Without including `{.list-icon}`, the following list style is created:

<img width="448" alt="Screen Shot 2023-08-24 at 9 57 46 PM" src="https://github.com/retypeapp/retype/assets/62210/c53109ea-2e38-45f1-a4eb-d38041f85efa">

Once again, this new `list-icon` style will be available in the upcoming Retype v3.3 release. 

Hope this helps.

**Comment by geoffreymcgill** [comment]
Generic attribute support has been added to the List Item component and your original sample is now working as expected.

The original sample of applying a custom color to the list item marker is now supported:

```md
<style>
li.item-green::marker {
    color: green;
}

li.item-yellow::marker {
    color: yellow;
}

li.item-red::marker {
    color: red;
}
</style>

- Item 1 (green) {.item-green}
- Item 2 (yellow) {.item-yellow}
- Item 3 (red) {.item-red}
```

![Image](https://github.com/user-attachments/assets/914f89a4-542b-4943-bb11-1423f7ec63b7)

The following demo is similar to the above but simplifies the styles and applies the color to the entire list item text:

```md
<style>
.item-green {
    color: green;
}

.item-yellow {
    color: yellow;
}

.item-red {
    color: red;
}
</style>

- Item 1 (green) {.item-green}
- Item 2 (yellow) {.item-yellow}
- Item 3 (red) {.item-red}
```

![Image](https://github.com/user-attachments/assets/df51fe43-e83a-4b08-8728-1e1605f62079)

The new functionality will be included in the upcoming `v3.10.0` release.

Hope this helps.
